phases/13-tools-and-protocols/09-mcp-transports/code/main.py

```
@@ -0,0 +1,229 @@
+            print(f"  -> HTTP {resp.status} (unexpected)")
+    except urllib.error.HTTPError as e:
+        print(f"  -> HTTP {e.code} (expected 404)")
+
+    srv.shutdown()
+
+
+def main() -> None:
+    if len(sys.argv) > 1 and sys.argv[1] == "--probe":
+        probe()
+        return
+    srv = serve("0.0.0.0", 8017)
```

Ruff (0.15.11)

[error] Possible binding to all interfaces

(S104)